pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -410,10 +410,37 @@ func translateRoute(push *model.PushContext, node *model.Proxy, in *networking.H
 
 			hostname := model.Hostname(dst.GetDestination().GetHost())
 			n := GetDestinationCluster(dst.Destination, serviceRegistry[hostname], port)
-			weighted = append(weighted, &route.WeightedCluster_ClusterWeight{
-				Name:   n,
-				Weight: weight,
-			})
+			clusterWeight := &route.WeightedCluster_ClusterWeight{
+				Name:                    n,
+				Weight:                  weight,
+				ResponseHeadersToRemove: dst.RemoveResponseHeaders,
+			}
+
+			if len(dst.AppendHeaders) > 0 {
+				clusterWeight.RequestHeadersToAdd = make([]*core.HeaderValueOption, 0)
```

Comment:
this check is unnecessary. `append` handles the `clusterWeight.RequestHeadersToAdd == nil` case